### ISSUE TITLE
Incorrect Display of Network Addresses for CIDR /31 and /32 and Subnets 255.255.255.254 and 255.255.255.255

### ISSUE THREAD

**Issue by HRA42** [HRA42/Go-IP#2]
We've observed an inconsistency in the representation of network values for higher CIDR notations, specifically CIDR /31 and /32. Additionally, the corresponding subnet masks of 255.255.255.254 and 255.255.255.255 seem to display these values inaccurately. This can potentially lead to configuration errors and misinterpretation of network boundaries. A prompt review and rectification would greatly enhance the accuracy and reliability of the system.

From the logs below, note the following anomalies:

For a /31 CIDR, the last usable address is shown as the network address itself, which is incorrect.
For a /32 CIDR, the first usable address should be the same as the network address, but it's being shown as one address higher. Additionally, the last usable address is shown as 192.168.1.255, which is not applicable for a /32 CIDR notation.

```bash
Bitte gib die Netzwerk Adresse ein: 192.168.1.1/31
Du hast folgende Netzwerk Adresse eingegeben: 192.168.1.0/31
2023/08/16 20:45:22 Die Subnetzmaske ist: 255.255.255.254
2023/08/16 20:45:22 Die Netzwerk Adresse ist: 192.168.1.0
2023/08/16 20:45:22 Die erste nutzbare Adresse des Netzwerks ist: 192.168.1.1
2023/08/16 20:45:22 Die letzte nutzbare Adresse des Netzwerks ist: 192.168.1.0
2023/08/16 20:45:22 Die Broadcast Adresse des Netzwerks ist: 192.168.1.1
```

```bash
Bitte gib die Netzwerk Adresse ein: 192.168.1.0/32
Du hast folgende Netzwerk Adresse eingegeben: 192.168.1.0/32
2023/08/16 20:45:55 Die Subnetzmaske ist: 255.255.255.255
2023/08/16 20:45:55 Die Netzwerk Adresse ist: 192.168.1.0
2023/08/16 20:45:55 Die erste nutzbare Adresse des Netzwerks ist: 192.168.1.1
2023/08/16 20:45:55 Die letzte nutzbare Adresse des Netzwerks ist: 192.168.1.255
2023/08/16 20:45:55 Die Broadcast Adresse des Netzwerks ist: 192.168.1.0
```

**Comment by HRA42** [comment]
Corresponding Functions:

Line 30 on network package
```go
func (n Network) GetFirstIP() net.IP {
	// TODO: Fix for CIDR 31 and 32 / SubnetMask 255.255.255.254 and 255.255.255.255
	networkAddress := n.GetNetworkAddress().To4()
	firstUsableAddress := make(net.IP, len(networkAddress))
	copy(firstUsableAddress, networkAddress)
	firstUsableAddress[3]++
	return firstUsableAddress
}
``` 

Line 39 on network package
```go
func (n Network) GetFirstIP() net.IP {
	// TODO: Fix for CIDR 31 and 32 / SubnetMask 255.255.255.254 and 255.255.255.255
	networkAddress := n.GetNetworkAddress().To4()
	firstUsableAddress := make(net.IP, len(networkAddress))
	copy(firstUsableAddress, networkAddress)
	firstUsableAddress[3]++
	return firstUsableAddress
}
```

**Comment by HRA42** [comment]
Sorry, marked a wrong PR for this.

This needs a better approach then just incrementing and decrementing the network or broadcast address. Will look into it tomorrow.
